pybamm/parameters/parameter_values.py

```diff
@@ -181,13 +181,21 @@ def process_symbol(self, symbol):
         elif isinstance(symbol, pybamm.FunctionParameter):
             new_child = self.process_symbol(symbol.children[0])
             function_name = self.get_parameter_value(symbol)
-            function = pybamm.Function(pybamm.load_function(function_name), new_child)
+            function = pybamm.load_function(function_name)
+            # It's possible to define a FunctionParameter with a function name that
+            # just returns a scalar (i.e. isn't really a function). For example, the
+            # constant current function is just equal to 1.
+            if not callable(function):
```

Comment:
not sure I understand this. The function
```python
def return_a_constant():
    return 1
```
is still callable, isn't it?

Reply:
couldn't you just call `pybamm.Symbol.is_constant()` on the FunctionParameter node?

Reply:
the issue is that the function (the function that we don't know whether it's callable or not) is only called by `Function.evaluate()`, so while we are building the tree we don't know whether the function depends on y or t. In the expression tree, `Function` is a node with a child (`t` in the case of current), regardless of what its function does. 

The function 
```python3
def return_a_constant(t):
    return 1
```
 is callable, but I don't know a way of checking that it always returns a constant without either using a flag or testing for an arbitrary number of values of t. Is there something in the function signature that could tell us?

Reply:
but we do know if the function depends on y or t. This FunctionParameter depends on t:

```python
f = pybamm.FunctionParameter('depends_on_t', pybamm.t)
```

This one depends on the parameter `p`:

```python
p = pybamm.Parameter('p')
f = pybamm.FunctionParameter('depends_on_p', p)
```

The only way the function can depend on either y or t is if `pybamm.Variable` or `pybamm.t` are in the expression tree below the `FunctionParameter`

Reply:
actually, looking at the `simplify` function for `pyamm.Function`, this simplification should already be taking place??

Reply:
But the Function
```python3
f = pybamm.Function("return_a_constant", pybamm.t)
```
depends on t in the "expression tree" sense, but when evaluated it will be equal to 1 regardless of the value of t. But `f.is_constant()` is false, since the expression tree thinks that f depends on t. In this case, we should have just defined `f=Scalar(1)`, but we don't know that when we are creating the model

Reply:
ahhh, ok. I get it now. That is a tricky one. 

you could use the inspect module (https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object).

The constant function could then look like this:
```
def constant():
    return 1
```

You can use the inspect module to count the number of arguments of the given function. If its 0, then you can simplify, if its 1, then you can't, if its > 1 then you raise an error. What do you think? 

The trouble with what you have done, is that say the user makes a mistake with their function and it not callable (when it should be), then pybamm silently simplifies it away.....